TestValidator.test_bin_dir_is_file failed in my environment.

### What happened?

```
_________________________ TestValidator.test_bin_dir_is_file _________________________

self = <tests.test_validator.TestValidator testMethod=test_bin_dir_is_file>
mock_out = <_io.StringIO object at 0x7f1354c14d30>
mock_err = <_io.StringIO object at 0x7f13517e43a0>

    def test_bin_dir_is_file(self, mock_out, mock_err):
        files.append(config["postgresql"]["data_dir"])
        files.append(config["postgresql"]["bin_dir"])
        c = copy.deepcopy(config)
        c["restapi"]["connect_address"] = 'False:blabla'
        c["postgresql"]["listen"] = '*:543'
        c["etcd"]["hosts"] = ["127.0.0.1:2379", "1244.0.0.1:2379", "127.0.0.1:invalidport"]
        c["kubernetes"]["pod_ip"] = "127.0.0.1111"
        errors = schema(c)
        output = "\n".join(errors)
>       self.assertEqual(['etcd.hosts.1', 'etcd.hosts.2', 'kubernetes.pod_ip', 'postgresql.bin_dir',
                          'postgresql.data_dir', 'raft.bind_addr', 'raft.self_addr',
                          'restapi.connect_address'], parse_output(output))
E       AssertionError: Lists differ: ['etcd.hosts.1', 'etcd.hosts.2', 'kubernetes.pod_ip', [102 chars]ess'] != ['etcd.hosts.2', 'kubernetes.pod_ip', 'postgresql.bin_[86 chars]ess']
E
E       First differing element 0:
E       'etcd.hosts.1'
E       'etcd.hosts.2'
E
E       First list contains 1 additional elements.
E       First extra element 7:
E       'restapi.connect_address'
E
E       - ['etcd.hosts.1',
E       ?              ^
E
E       + ['etcd.hosts.2',
E       ?              ^
E
E       -  'etcd.hosts.2',
E          'kubernetes.pod_ip',
E          'postgresql.bin_dir',
```

### How can we reproduce it (as minimally and precisely as possible)?

#### OS
```
vagrant@citus:/citus_cluster/patroni$ cat /etc/os-release
PRETTY_NAME="Ubuntu 22.04.2 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.2 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=jammy
```

#### Patroni
master branch, commit id: c96e35c807fcf1f8598612761f2aaa4acb644590

### What did you expect to happen?

I will create a pull request if this is a bug.

### Patroni/PostgreSQL/DCS version

-   Patroni version: master
-   PostgreSQL version: seems not related
-   DCS (and its version): seems not related


### Patroni configuration file

```yaml
not related
```


### patronictl show-config

```yaml
not related
```


### Patroni log files

```shell
not related
```


### PostgreSQL log files

```shell
not related
```


### Have you tried to use GitHub issue search?

- [X] Yes

### Anything else we need to know?

_No response_

--- a/tests/test_validator.py
+++ b/tests/test_validator.py
@@ -134,6 +134,23 @@ def connect_side_effect(host_port):
         raise socket.gaierror()
 
 
+def mock_getaddrinfo(host, port, *args):
+    if port is None or port == "":
+        port = 0
+    port = int(port)
+    if port not in range(0, 65536):
+        raise socket.gaierror()
+
+    if host == "127.0.0.1" or host == "" or host is None:
+        return [(socket.AF_INET, socket.SOCK_STREAM, socket.IPPROTO_TCP, '', ('127.0.0.1', port))]
+    elif host == "127.0.0.2":
+        return [(socket.AF_INET, socket.SOCK_STREAM, socket.IPPROTO_TCP, '', ('127.0.0.2', port))]
+    elif host == "::1":
+        return [(socket.AF_INET6, socket.SOCK_STREAM, socket.IPPROTO_TCP, '', ('::1', port, 0, 0))]
+    else:
+        raise socket.gaierror()
+
+
 def parse_output(output):
     result = []
     for s in output.split("\n"):
@@ -145,6 +162,7 @@ def parse_output(output):
 
 
 @patch('socket.socket.connect_ex', Mock(side_effect=connect_side_effect))
+@patch('socket.getaddrinfo', Mock(side_effect=mock_getaddrinfo))
 @patch('os.path.exists', Mock(side_effect=exists_side_effect))
 @patch('os.path.isdir', Mock(side_effect=isdir_side_effect))
 @patch('os.path.isfile', Mock(side_effect=isfile_side_effect))

